src/api/buddies.ts

```diff
@@ -26,6 +32,9 @@ export type BanStatusStrings = 'banned' | 'ok' | 'deleted';
 type MappedStatuses = {
   [Action in BanActions]: BanStatusStrings;
 };
+type MappedStatusesFromApi = {
+  [K in BanStatusStrings]: BanStatuses;
+};
```

Comment:
Record would be maybe more readable.

`Record<BanStatusStrings, BanStatuses>`

aaaah. this is not what you want .this type is borken. if you want a mapper just use "as const"

`const toBanStatus = {
  banned: 'Banned',
  deleted: 'Deleted',
  ok: 'NotBanned',
} as const`

Reply:
the difference is that the current type creates 3x3 possibilities but we want 1:1 mapping.

Reply:
you can then derive the types from the const declared thingy by typeof and [] in type level if you want to be super DRY.

Reply:
Is union type in plural or in singular?

I personally have favored non-plurals. But I don't have any argument why.

Reply:
Yeah, I also like unions as singular. I went ahead and changed all to singular form. =)